sympy/stats/tests/test_stochastic_process.py

```diff
@@ -652,7 +655,7 @@ def test_GammaProcess_symbolic():
     raises(IndexError, lambda: X(-1))
     assert isinstance(X(t), RandomIndexedSymbol)
     assert X.state_space == Interval(0, oo)
-    assert X.distribution(X(t)) == GammaDistribution(g*t, 1/l)
+    assert X.distribution(t) == GammaDistribution(g*t, 1/l)
```

Comment:
This isn't private API. Add a deprecation warning in `distribution`.
Possible add a function, like, `_deprecation_warn_distribution(self)` in the root level class and write the deprecation warning code inside it. Then call that function in the distribution method of each subclass.